kimchi/src/circuits/lookup/index.rs

```diff
@@ -185,12 +185,14 @@ impl<F: FftField + SquareRootField> LookupConstraintSystem<F> {
         runtime_tables: Option<Vec<RuntimeTableCfg<F>>>,
         domain: &EvaluationDomains<F>,
     ) -> Result<Option<Self>, LookupError> {
-        let lookup_info = LookupInfo::create(runtime_tables.is_some());
-
         //~ 1. If no lookup is used in the circuit, do not create a lookup index
-        match lookup_info.lookup_used(gates) {
+        match LookupInfo::create_from_gates(gates, runtime_tables.is_some()) {
             None => Ok(None),
-            Some(lookup_used) => {
+            Some(lookup_info) => {
```

Comment:
to avoid all the indented code, this should be replaced with

```rust
let runtime_tables_used = runtime_tables.is_some();
let lookup_info = LookupInfo::create_from_gates(gates, runtime_tables_used).ok_or(...)?;
// ...
```